File: app/src/main/java/com/originb/inkwisenote2/modules/repositories/NoteRelationRepository.java
package com.originb.inkwisenote2.modules.repositories;

import com.originb.inkwisenote2.modules.ocr.data.NoteOcrTextsDao;
import com.originb.inkwisenote2.modules.ocr.data.NoteTermFrequencyDao;
import com.originb.inkwisenote2.modules.noterelation.data.NoteRelationDao;
import com.originb.inkwisenote2.modules.smartnotes.data.AtomicNoteEntity;

public class NoteRelationRepository {
    private final NoteRelationDao noteRelationDao;
    private final NoteOcrTextsDao noteOcrTextDao;
    private final NoteTermFrequencyDao noteTermFrequencyDao;

    public NoteRelationRepository(NoteRelationDao noteRelationDao,
                                  NoteOcrTextsDao noteOcrTextDao,
                                  NoteTermFrequencyDao noteTermFrequencyDao) {
        this.noteRelationDao = noteRelationDao;
        this.noteOcrTextDao = noteOcrTextDao;
        this.noteTermFrequencyDao = noteTermFrequencyDao;
    }

    public void deleteNoteRelationData(AtomicNoteEntity atomicNote) {
        noteRelationDao.deleteByNoteId(atomicNote.getNoteId());
        noteOcrTextDao.deleteNoteText(atomicNote.getNoteId());
        noteTermFrequencyDao.deleteTermFrequencies(atomicNote.getNoteId());
    }
}

File: app/src/main/java/com/originb/inkwisenote2/modules/repositories/NoteRelationRepository.kt
package com.originb.inkwisenote2.modules.repositories

import com.originb.inkwisenote2.modules.noterelation.data.NoteRelationDao
import com.originb.inkwisenote2.modules.ocr.data.NoteOcrTextDao
import com.originb.inkwisenote2.modules.ocr.data.NoteTermFrequencyDao
import com.originb.inkwisenote2.modules.smartnotes.data.AtomicNoteEntity

class NoteRelationRepository {
    private val noteRelationDao: NoteRelationDao =
        Repositories.Companion.getInstance().getNotesDb().noteRelationDao()
    private val noteOcrTextDao: NoteOcrTextDao =
        Repositories.Companion.getInstance().getNotesDb().noteOcrTextDao()
    private val noteTermFrequencyDao: NoteTermFrequencyDao =
        Repositories.Companion.getInstance().getNotesDb().noteTermFrequencyDao()

    fun deleteNoteRelationData(atomicNote: AtomicNoteEntity?) {
        noteRelationDao.deleteByNoteId(atomicNote.getNoteId())
        noteOcrTextDao.deleteNoteText(atomicNote.getNoteId())
        noteTermFrequencyDao.deleteTermFrequencies(atomicNote.getNoteId())
    }
}
